aws/warehouse.go

```diff
@@ -266,13 +383,13 @@ func (wh *AccountWarehouse) MintTokenWithTTL(ctx context.Context, params *Resour
 		return nil, err
 	}
 
-	return wh.ensureTokenResult(sess, principalArn, princSpec)
+	return wh.ensureTokenResult(sess, principalArn, princSpec, ctx, svcUserArn)
 }
 
-func (wh *AccountWarehouse) ensureTokenResult(sess *session.Session, principalArn string, princSpec *principalSpec) (*clouds.AwsResourceTokenResult, error) {
+func (wh *AccountWarehouse) ensureTokenResult(sess *session.Session, principalArn string, princSpec *principalSpec, ctx context.Context, svcUserArn string) (*clouds.AwsResourceTokenResult, error) {
```

Comment:
I've noticed that when the `Context` is in an argument list, it tends to be the first argument. It might be better stylistically to do that refactoring.